### PR TITLE
PERF-28, PERF-31 Model and create JMETER script

### PR DESCRIPTION
**Pull request by Svitlana-Zmiivska**
**Purpose:**
https://issues.folio.org/browse/PERF-28
https://issues.folio.org/browse/PERF-31

**Approach:**
Investigate slow queries by writing JMeter scripts:
- inventory_saveUUIDsInstance.jmx - validate slow queries in Inventory->Instance App->Save instances UUIDs

**TODO:** 
- Write test reports

### CODE REVIEW

File: workflows-scripts/item-search-filter/inventory_saveUUIDsInstance.jmx [hunk 1]
```diff
@@ -0,0 +1,1722 @@
...
+                <stringProp name="-64778490">&quot;instances&quot;</stringProp>
+              </collectionProp>
+              <stringProp name="Assertion.custom_message"></stringProp>
+              <stringProp name="Assertion.test_field">Assertion.response_data</stringProp>
+              <boolProp name="Assertion.assume_success">false</boolProp>
+              <intProp name="Assertion.test_type">2</intProp>
+            </ResponseAssertion>
+            <hashTree/>
+          </hashTree>
+          <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="GET_instance-bulk/ids" enabled="true">
```
Comment (mtraneis) [inline workflows-scripts/item-search-filter/inventory_saveUUIDsInstance.jmx:1589]: This call sends back the JSON of IDs but also with the csv file as well. Not sure if you can mimic the call in JMeter such that the response is also a csv file? I don't know if the server is taking more time to create the csv file or not.

File: workflows-scripts/item-search-filter/README.md [hunk 2]
```diff
@@ -18,6 +18,12 @@
     UC/HP/AANet/Intrnet or UC/HP/ASR/ACASA or UC/HP/ASR/ARCHASR or UC/HP/ASR/ASRHP or UC/HP/ASR/Atk or bUC/HP/ASR/GameASR or UC/HP/ASR/HarpASR or UC/HP/ASR/JRLASR or UC/HP/ASR/LawASR
    These locations are from the PTF perf testing environment (database)
 
+##### Workflow for inventory_saveUUIDsInstance.jmx
+1. Open Inventory App
+2. Select Instance tab in top left
+3. Select Query search in the drop down and type "keyword all "Boston" and source=FOLIO" in search field
+4. Under Actions menu click the option “Save instances UUIDs”
```
Comment (mtraneis) [inline workflows-scripts/item-search-filter/README.md:25]: To model this as closely to a real user behavior as possible, can a delay be added between landing on the inventory page and making a search, and before calling the export API? See the check-in/check-out workflow for examples of adding a delay.

File: workflows-scripts/item-search-filter/inventory_saveUUIDsInstance.jmx (hunk 1 above)
Comment (Svitlana-Zmiivska) [reply workflows-scripts/item-search-filter/inventory_saveUUIDsInstance.jmx:1589]: The backend API provides JSON with instance UUIDs, and not a file. UI receives a response containing JSON and convert it to the file.  That why I check only json response. I amn't sure if we can check a csv file

File: workflows-scripts/item-search-filter/jmeter-supported-data/search_queries.csv [hunk 3]
```diff
@@ -0,0 +1 @@
+keyword all "Boston" and source=FOLIO
```
[the remark this answers is not in the document]
Comment (Svitlana-Zmiivska) [reply workflows-scripts/item-search-filter/jmeter-supported-data/search_queries.csv]: That makes sense to me. For now I've added 10 queries. I am looking to add more queries

**Comment by Svitlana-Zmiivska** [comment]
@mtraneis the folder structures have been reorganized

File: workflows-scripts/inventory/exporting-inventory-search/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+1. Open .jmx file in JMeter IDE
+2. Change credentials in jmeter-supported-data\credentials.csv 
+3. Change hostname in User Defined Variables
+4. Click to the play button
+
+##### Using non-GUI:
+Navigate to Apache bin folder and run script.
+For example:
+/apache-jmeter-5.2.1/bin                                              
+▶ sudo ./jmeter.sh -n -t [path_to_project]/perf-testing/workflows-scripts/item-search-filter/users_search.jmx -l [path_to_project]/perf-testing/workflows-scripts/item-search-filter/result.jtl
```
Comment (javalkar-varun) [inline workflows-scripts/inventory/exporting-inventory-search/README.md:31]: After folder structure was modified, this path should be updated as well

File: workflows-scripts/inventory/instance-search-filter/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+1. Open .jmx file in JMeter IDE
+2. Change credentials in jmeter-supported-data\credentials.csv 
+3. Change hostname in User Defined Variables
+4. Click to the play button
+
+##### Using non-GUI:
+Navigate to Apache bin folder and run script.
+For example:
+/apache-jmeter-5.2.1/bin                                              
+▶ sudo ./jmeter.sh -n -t [path_to_project]/perf-testing/workflows-scripts/item-search-filter/users_search.jmx -l [path_to_project]/perf-testing/workflows-scripts/item-search-filter/result.jtl
```
Comment (javalkar-varun) [inline workflows-scripts/inventory/instance-search-filter/README.md:31]: After folder structure was modified, this path should be updated as well

File: workflows-scripts/inventory/exporting-inventory-search/inventory_saveUUIDsInstance.jmx [hunk 6]
```diff
@@ -0,0 +1,1751 @@
...
+              <intProp name="Assertion.test_type">2</intProp>
+            </ResponseAssertion>
+            <hashTree/>
+            <UniformRandomTimer guiclass="UniformRandomTimerGui" testclass="UniformRandomTimer" testname="Uniform Random Timer" enabled="true">
+              <stringProp name="ConstantTimer.delay">5000 + ${thinktime_offset}</stringProp>
+              <stringProp name="RandomTimer.range">2000</stringProp>
+            </UniformRandomTimer>
+            <hashTree/>
+          </hashTree>
+          <HTTPSamplerProxy guiclass="HttpTestSampleGui" testclass="HTTPSamplerProxy" testname="GET_/instance-bulk/ids" enabled="true">
```
Comment (javalkar-varun) [inline workflows-scripts/inventory/exporting-inventory-search/inventory_saveUUIDsInstance.jmx:1604]: Running script against perf env `okapi-fcp1-us-east-1.int.aws.folio.org`, tests are failing
image

**Comment by mtraneis** [comment]
> @mtraneis the folder structures have been reorganized

Looks great! Thank you.

File: workflows-scripts/inventory/exporting-inventory-search/README.md (hunk 4 above)
Comment (Svitlana-Zmiivska) [reply workflows-scripts/inventory/exporting-inventory-search/README.md:31]: @javalkar-varun @mtraneis How about moving `Steps to run JMeter script:` to the README file int the root `perf-testing` folder or in the `workflows-scripts` folder? In order not to repeat these steps for each workflow

Comment (mtraneis) [reply workflows-scripts/inventory/exporting-inventory-search/README.md:31]: Agreed. Thanks

File: workflows-scripts/inventory/exporting-inventory-search/inventory_saveUUIDsInstance.jmx (hunk 6 above)
Comment (Svitlana-Zmiivska) [reply workflows-scripts/inventory/exporting-inventory-search/inventory_saveUUIDsInstance.jmx:1604]: It is expected behavior. It shows that after some time we can't get response. More details I'll describe in a test report

**Pull request merged**